cloud/centurylink/clc_server.py

```diff
@@ -0,0 +1,1373 @@
+        self.module.params = self._validate_module_params(
+            self.clc,
+            self.module)
+        p = self.module.params
+        state = p.get('state')
+
+        #
+        #  Handle each state
+        #
+        partial_servers_ids = []
+        if state == 'absent':
+            server_ids = p['server_ids']
+            if not isinstance(server_ids, list):
+                return self.module.fail_json(
+                    msg='server_ids needs to be a list of instances to delete: %s' %
+                    server_ids)
+
+            (changed,
+             server_dict_array,
+             new_server_ids) = self._delete_servers(module=self.module,
+                                                    clc=self.clc,
+                                                    server_ids=server_ids)
+
+        elif state in ('started', 'stopped'):
+            server_ids = p.get('server_ids')
+            if not isinstance(server_ids, list):
+                return self.module.fail_json(
+                    msg='server_ids needs to be a list of servers to run: %s' %
+                    server_ids)
+
+            (changed,
+             server_dict_array,
+             new_server_ids) = self._start_stop_servers(self.module,
+                                                        self.clc,
+                                                        server_ids)
+
+        elif state == 'present':
+            # Changed is always set to true when provisioning new instances
+            if not p.get('template'):
+                return self.module.fail_json(
+                    msg='template parameter is required for new instance')
+
+            if p.get('exact_count') is None:
+                (server_dict_array,
+                 new_server_ids,
+                 partial_servers_ids,
+                 changed) = self._create_servers(self.module,
+                                                 self.clc)
+            else:
+                (server_dict_array,
+                 new_server_ids,
+                 partial_servers_ids,
+                 changed) = self._enforce_count(self.module,
+                                                self.clc)
+
+        self.module.exit_json(
+            changed=changed,
+            server_ids=new_server_ids,
+            partially_created_server_ids=partial_servers_ids,
+            servers=server_dict_array)
+
+    @staticmethod
+    def _define_module_argument_spec():
+        """
+        Define the argument spec for the ansible module
+        :return: argument spec dictionary
+        """
+        argument_spec = dict(
+            name=dict(),
+            template=dict(),
+            group=dict(default='Default Group'),
+            network_id=dict(),
+            location=dict(default=None),
+            cpu=dict(default=1),
+            memory=dict(default=1),
+            alias=dict(default=None),
+            password=dict(default=None),
```

Comment:
Thank you @resmo for the comment. I have added the no_log attribute to password argument.